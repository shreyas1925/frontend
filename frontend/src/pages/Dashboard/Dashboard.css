.dashboard--container{
    display: flex;
   min-height: 100vh;
    width: 100%;
    background-color: #f0f2ff;
}

.dashboard--content{
    width: 75%;
    height: 100%;
    display: flex;
}

.dashboard--content--left{
    width: 35%;
    min-height: 100vh;
    background-color: #d4d6e5;
}

.dashboard--content--right{
    width: 65%;
    height: 100%;
    background-color: #eaeefe;
}
.header{
    background-color: #ffffff;
    height: 50px;
    padding: 10px;
    line-height: 50px;
}

.header h1{
    color:black;
    font-size: 25px;
    font-family: "Roboto", sans-serif;
    font-weight: 500;
}

.dashboard--header{
    background-color: #ffffff;
    height: 50px;
    padding: 10px;
    line-height: 50px;
    font-size: 25px;
    font-family: "Roboto", sans-serif;
    font-weight: 500;
}

.dashboard--types{
    color:white;
    display: flex;
    font-size: 10px;
    justify-content: center;
    margin-top: 40px;
    display: flex;
    justify-content: space-around;
}

.dashboard--types h1{
    font-family: "Roboto", sans-serif;
    text-transform: uppercase;
    font-size: 16px;
    color:#afafaf
}

.dashboard--types img{
    width: 25px;
    height: 25px;
    margin-right: 0px;
    margin-top: -6px;
    padding-left:100px ;
}

.content-type--button{
    border:1px dotted #5906ce;
    width: 220px;
    margin-left: 42px;
    margin-top: -70px;
    padding: 16px;
    padding-left: 40px;
    color:#5906ce;
    text-align: center;
    font-family: "roboto", sans-serif;
    cursor: pointer;
}
.content-list{
    background-color: white;
    width: 200px;
    border-radius: 8px;
    margin-left: 42px;
    margin-top:30px;
    padding: 16px;
    padding-left: 60px;
    color:#afafaf;
    font-family: "roboto", sans-serif;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
}

.content-list h2{
    font-size: 16px;
    font-weight: 500;
}

.content-list p{
    font-size: 16px;
    font-weight: 500;
    color:#afafaf;
}

.content-list:hover{
    background-color: #5906ce;
    width: 200px;
    border-radius: 8px;
    margin-left: 42px;
    margin-top:30px;
    padding: 16px;
    padding-left: 60px;
    color:#fff;
    font-family: "roboto", sans-serif;
    cursor: pointer;
    transition: all 0.4s;
}

.content--header{
    margin-top: 30px;
    margin-left: 30px;
    margin-right: 30px;
    display: flex;
}

.content--header h1{
    font-size: 20px;
    font-weight: 800;
    color:black;
    font-family: "roboto", sans-serif;
}

.content--header p{
    margin-top: -80px;
    margin-left: 0px;
}

.content--header img{
    width: 20px;
    height: 20px;
    margin-left: 10px;
}

.content-type--btn{
    border:1px dotted #5906ce;
    width: 550px;
    margin-left: 30px;
    margin-top: 37px;
    padding: 16px;
    padding-left: 60px;
    color:#5906ce;
    font-family: "roboto", sans-serif;
    cursor: pointer;
    text-align: center;
}

.content--field{
    border:none;
    width: 580px;
    margin-left: 30px;
    margin-top: 37px;
    color:black;
    font-family: "roboto", sans-serif;
    cursor: pointer;
    text-align: center;
    background-color: white;
    display: flex;
    justify-content: space-between;
    border-radius: 8px;
    padding: 24px;
}
.content--field h2,h3{
    font-size: 18px;
    font-weight: 500;
    color:#afafaf;
}

.content--field h2{
    margin-left: 30px;
}

.content--field h3{
    margin-right: 25px;
}

.string{
    position: relative;
}
.string p{
    background-color: #7791fe;
    color:white;
    width: 55px;
    position:absolute;
    top: 38px;
    line-height: 70px;
    border-radius: 8px 0px 0px 8px;
    height: 72px;
    text-align: center;

}

.contents{
    display: flex;
    padding: 16px;
    justify-content: space-around;
}

.contents h2{
    font-size: 16px;
}

.images{
    width:15px;
    height: 15px;
    display: flex;
    margin-right:40px ;
    justify-content: space-between;
}

.images img{
    width: 15px;
    height: 15px;
    margin-left:20px;
}

@media screen and (max-width:900px)
{
    .dashboard--container{
        display: flex;
        flex-direction: column;
        min-height: 100vh;
        width: 100%;
        background-color: #f0f2ff;
    }

    .dashboard--content{
        width: 100%;
        height: 100%;
        display: flex;
        flex-direction: column;
    }
    .dashboard--content--left{
        width: 100%;
        min-height: 100vh;
        background-color: #d4d6e5;
    }
    
    .dashboard--content--right{
        width: 100%;
        height: 100%;
        background-color: #eaeefe;
    }

    .content-type--btn{
        margin:30px;
        margin-bottom: 40px;
    }

    .string{
        margin-top:60px;
    }
    .string p{
        margin-top:-48px;
    }
}
    
   