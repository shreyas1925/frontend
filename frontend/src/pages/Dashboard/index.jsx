import React from "react";
import "./Dashboard.css";
import { Viewbar } from "../../components";
import icon from "../../assets/icon.png";
import axios from 'axios'
import pencil from '../../assets/pencil.png'
import edit from '../../assets/edit.png'
import deleteIcon from '../../assets/delete.png'
import Modal from "../../components/Modal";

const Dashboard = () => {
  const [showModal, setShowModal] = React.useState(false);
  const [contents,setContents] = React.useState([])
  React.useEffect(() => {
    axios.get('http://localhost:5000/api/getContents').then(res => {
      console.log(res);
      setContents(res.data)
    }).catch(err => {
      console.log(err);
    })
    },[])

  return (
    <div className="dashboard--container">
      <Viewbar contents={contents}/>
      
      <div className="dashboard--content">
        
        <div className="dashboard--content--left">
        <div className="header">
            <h1>Content Types</h1>
        </div>
             
        <div className='dashboard--types'>
          <h1>7 Types</h1>
          <img src={icon} alt="" />
        </div>

          <div className="content-type--button" onClick={() => setShowModal(true)}>
            + New Type
          </div>

          <Modal onClose={() => setShowModal(false)} show={showModal} />
          
            {
              contents.map((content)=>{
                return (
                  <div className="content-list">
                    <h2>{content.name}</h2>
                    <p>13</p>
                  </div>
                )
              })
            }
            
        </div>

        <div className="dashboard--content--right">
        <div className='dashboard--header'>
            
        </div>
        <div className="content--header">
         <div>
         <h1>Company profile</h1>
         <p>13 Fields</p>
          </div> 
          <img src={pencil} alt="" />
        </div>
        
        <div className="content-type--btn">
            Add Another field
        </div>

        {
          contents.map((content)=>{
            return (
              <>
                <div className="string">
                  <p>Ab</p> 
                </div>
           
              <div className="content--field">
                {/* <div className="contents"> */}
                  <div>
                    <h2>name</h2>
                  </div>
                  <div>
                    <h3>String</h3>
                  </div>
                
                  <div className="images">
                    <img src={edit} alt="" />
                    <img src={deleteIcon} alt="" />
                  </div>
                </div>
              {/* </div> */}
              </>
            )
          })
        }

        </div>
      </div>
    </div>
  );
};

export default Dashboard;
