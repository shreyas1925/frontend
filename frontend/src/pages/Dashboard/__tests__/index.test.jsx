import { BrowserRouter } from "react-router-dom";
import Dashboard from "..";
import { render } from "@testing-library/react";

describe("Dashboard", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<BrowserRouter> <Dashboard /></BrowserRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render and have text", () => {
    const { getByText } = render(<BrowserRouter> <Dashboard /></BrowserRouter>);
    expect(getByText("+ New Type")).toBeInTheDocument();
  })

  
});
