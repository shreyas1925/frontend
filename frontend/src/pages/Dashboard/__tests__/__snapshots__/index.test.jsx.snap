// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render correctly 1`] = `
<DocumentFragment>
   
  <div
    class="dashboard--container"
  >
    <div
      class="viewbar--container"
    >
      <div
        class="viewbar--header"
      >
        CMS+
      </div>
      <div
        class="viewbar--types"
      >
        <h1>
          Collection Types
        </h1>
        <img
          alt=""
          src="icon.png"
        />
      </div>
      <div
        class="viewbar--contents"
      />
      <div
        class="builder"
      >
        <h2>
           Content type builder
        </h2>
      </div>
    </div>
    <div
      class="dashboard--content"
    >
      <div
        class="dashboard--content--left"
      >
        <div
          class="header"
        >
          <h1>
            Content Types
          </h1>
        </div>
        <div
          class="dashboard--types"
        >
          <h1>
            0 Types
          </h1>
          <img
            alt=""
            src="icon.png"
          />
        </div>
        <div
          class="content-type--button"
        >
          + New Type
        </div>
      </div>
      <div
        class="dashboard--content--right"
      >
        <div
          class="dashboard--header"
        />
        <div
          class="content--header"
        >
          <div>
            <h1>
              Company_Profile
            </h1>
            <p />
          </div>
          <img
            alt=""
            src="pencil.png"
          />
        </div>
        <div
          class="content-type--btn"
        >
          Add Another field
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
